lib/stm32/l4/rcc.c

```diff
@@ -38,12 +38,51 @@
 /**@{*/
 #include <libopencm3/cm3/assert.h>
 #include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/stm32/pwr.h>
 
 /* Set the default clock frequencies after reset. */
 uint32_t rcc_ahb_frequency = 4000000;
 uint32_t rcc_apb1_frequency = 4000000;
 uint32_t rcc_apb2_frequency = 4000000;
 
+const struct rcc_clock_scale rcc_hsi16_configs[RCC_CLOCK_CONFIG_END] = {
+	{ /* 80MHz PLL from HSI16 VR1 */
+		.pllm = 4,
+		.plln = 40,
+		.pllp = RCC_PLLCFGR_PLLP_DIV7,
+		.pllq = RCC_PLLCFGR_PLLQ_DIV6,
+		.pllr = RCC_PLLCFGR_PLLR_DIV2,
+		.pll_source = RCC_PLLCFGR_PLLSRC_HSI16,
+		.hpre = RCC_CFGR_HPRE_NODIV,
+		.ppre1 = RCC_CFGR_PPRE_NODIV,
+		.ppre2 = RCC_CFGR_PPRE_NODIV,
+		.voltage_scale = PWR_SCALE1,
+		.flash_config = FLASH_ACR_DCEN | FLASH_ACR_ICEN |
+		FLASH_ACR_LATENCY_4WS,
+		.ahb_frequency  = 80000000,
+		.apb1_frequency = 80000000,
+		.apb2_frequency = 80000000,
+	},
+	{ /* 72MHz PLL from HSI16 VR1 */
```

Comment:
80MHz is max speed, what is 72 for?  We generally try and avoid having too many of these defined, only max for each part really.  Is this intended for L476 and L475 that have the dwc_otg usb core, to be max core speed that allows a 48MHz USB clock from HSE?

Reply:
That was my main logic behind it.